src/UI/Art/SpriteArtist/SpriteMapView.java

```diff
@@ -244,8 +248,25 @@ private BufferedImage renderMap()
     // Draw units, buildings, trees, etc.
     drawUnitsAndMapObjects(mapGraphics, gameMap, animIndex);
 
+    ArrayList<GameOverlay> overlays = new ArrayList<GameOverlay>();
     // Apply any relevant map highlight.
-    mapArtist.drawHighlights(mapGraphics);
+    if( null != currentPath && null != currentActor )
+    {
+      for( WeaponModel w : currentActor.model.weapons )
+      {
+        Color edgeColor = OverlayArtist.SIEGE_FIRE_EDGE;
+        if( w.canFireAfterMoving )
+          edgeColor = OverlayArtist.MOBILE_FIRE_EDGE;
+        overlays.add(new GameOverlay(null,
+                     Utils.findLocationsInRange(gameMap, myGame.getCursorCoord(), w.minRange, w.maxRange),
+                     OverlayArtist.FIRE_FILL, edgeColor));
+      }
+    }
+    for( Commander co : myGame.commanders )
+    {
+      overlays.addAll(co.getMyOverlays(gameMap, co == gameMap.viewer));
```

Comment:
Given that `co.getMyOverlays()` just returns an empty collection if `co != gameMap.viewer`, why bother with a loop here?

Reply:
I have *plaaaaaaaaaaaaans*.